RecoHI/HiEvtPlaneAlgos/interface/LoadEPDB.h

```diff
@@ -0,0 +1,115 @@
+#ifndef RecoHI_HiEvtPlaneAlgos_LoadEPDB_h
+#define RecoHI_HiEvtPlaneAlgos_LoadEPDB_h
+
+// system include files
+#include <memory>
+#include <iostream>
+#include <string>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "DataFormats/HeavyIonEvent/interface/EvtPlane.h"
+#include "RecoHI/HiEvtPlaneAlgos/interface/HiEvtPlaneFlatten.h"
+#include "RecoHI/HiEvtPlaneAlgos/interface/HiEvtPlaneList.h"
+#include <vector>
+
+using namespace hi;
+
+class LoadEPDB {
+ public:
+
+  explicit LoadEPDB(edm::ESHandle<RPFlatParams> flatparmsDB_, HiEvtPlaneFlatten ** flat)
+  {
+    int Hbins;
+    int Obins;
+    int flatTableSize = flatparmsDB_->m_table.size();
+    genFlatPsi_ = kTRUE;
+    if(flatTableSize<flat[0]->getHBins()+2*flat[0]->getOBins()) {
+      genFlatPsi_ = kFALSE;
+    } else {
+      Hbins = flat[0]->getHBins();
+      Obins = flat[0]->getOBins();
+
+      for(int i = 0; i<flatTableSize; i++) {
+	const RPFlatParams::EP* thisBin = &(flatparmsDB_->m_table[i]);
+	for(int j = 0; j<NumEPNames; j++) {
+	  int indx = thisBin->RPNameIndx[j];
+	  if(indx<0||indx>=NumEPNames) {
+	    genFlatPsi_ = kFALSE;
+	    break;
+	  }
+	  if(indx>=0) {
+	    if(i<Hbins) {
+	      flat[indx]->setXDB(i, thisBin->x[j]);
+	      flat[indx]->setYDB(i, thisBin->y[j]);
+	    } else if(i>=Hbins && i<Hbins+Obins) {
+	      flat[indx]->setXoffDB(i - Hbins, thisBin->x[j]);
+	      flat[indx]->setYoffDB(i - Hbins, thisBin->y[j]);
+
+	    } else if (i>=Hbins+Obins && i<Hbins+2*Obins) {
+	      flat[indx]->setPtDB(i - Hbins- Obins, thisBin->x[j]);
+	      flat[indx]->setPt2DB(i - Hbins- Obins, thisBin->y[j]);
+	    }
+	  }
+	}
+      }
+      int cbins = 0;
+      while(flatTableSize>Hbins + 2*Obins + cbins) {
+	const RPFlatParams::EP* thisBin = &(flatparmsDB_->m_table[Hbins+2*Obins +cbins]);
+	double centbinning = thisBin->x[0];
+	int ncentbins = (int) thisBin->y[0]+0.01;
+	if(ncentbins==0) break;
+	for(int j = 0; j< ncentbins; j++) {
+	  const RPFlatParams::EP* thisBin = &(flatparmsDB_->m_table[Hbins+2*Obins +cbins+j+1]);
```

Comment:
just checking: "thisBin" here and 5 lines above are meant to be called the same name, right?